mesonbuild/compilers/detect.py

```diff
@@ -240,8 +240,9 @@ def _get_compilers(env: 'Environment', lang: str, for_machine: MachineChoice) ->
         # Return value has to be a list of compiler 'choices'
         compilers = [comp]
     else:
-        if not env.machines.matches_build_machine(for_machine):
-            raise EnvironmentException(f'{lang!r} compiler binary not defined in cross or native file')
+        # Cross compilers must NEVER EVER be taken from envvars.
```

Comment:
@Ericson2314 @ePirat seems like we're about to regress again, sigh.

Reply:
Yeah thanks for letting me know. We have been using env vars with Meson quite successfully and necessarily with NixOS/Nixpkgs for many releases of Meson now, and I am sure others are too.